src/core/Table/Table.tsx

```diff
@@ -287,13 +287,11 @@ export const Table = <
           const disabled = instance.rows.every((row) =>
             isRowDisabled?.(row.original),
           );
-          const checked =
-            instance.isAllRowsSelected ||
-            instance.rows.every(
-              (row) =>
-                instance.state.selectedRowIds[row.id] ||
-                isRowDisabled?.(row.original),
-            );
+          const checked = instance.initialRows.every(
```

Comment:
I can't find any info on `initialRows` in react-table docs. Do you have a link?

Reply:
There isn't. Just code:
https://github.com/tannerlinsley/react-table/blob/8ec25cabccf915991ec00cf67bfb26a83aa0c0b2/src/hooks/useTable.js#L213

Reply:
So it's a secret property that magically fixes our bug?

Reply:
Kind of. Typings come from different package so they might missed it.

Reply:
And did you narrow down why the crash was happening? Need to remove both `instance.rows` and `isAllRowsSelected`?

Reply:
Of course 😃 All selected rows are being saved to `selectedRowIds` even if table is filtered. Previously I was going through each selected row id and trying to get data from `rowsById` map. However, it only has filtered rows, so it tried to get `.original` from `undefined` row. 
I had to remove `isAllRowsSelected` because it is also checking only in `rowsById` map which is filtered and in order to achieve the same behavior as material-ui, I have to check each initial row.